gensim/similarities/nmslib.py

```diff
@@ -0,0 +1,217 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2019 Radim Rehurek <me@radimrehurek.com>
+# Copyright (C) 2019 Masahiro Kazama <kazama.masa@gmail.com>
+# Licensed under the GNU LGPL v2.1 - http://www.gnu.org/licenses/lgpl.html
+
+"""
+Intro
+-----
+This module contains integration Nmslib with :class:`~gensim.models.word2vec.Word2Vec`,
+:class:`~gensim.models.doc2vec.Doc2Vec`, :class:`~gensim.models.fasttext.FastText` and
+:class:`~gensim.models.keyedvectors.KeyedVectors`.
+
+
+What is Nmslib
+-------------
+Non-Metric Space Library (NMSLIB) is an efficient cross-platform similarity search library and a toolkit
+for evaluation of similarity search methods. The core-library does not have any third-party dependencies.
+
+
+How it works
+------------
+Searching in generic non-metric space.
+
+More information about Nmslib: `github repository <https://github.com/nmslib/nmslib>`_.
+
+"""
+import os
+
+from smart_open import smart_open
```

Comment:
Please use smart_open's open function instead.

See https://github.com/RaRe-Technologies/smart_open#migrating-to-the-new-open-function

Reply:
Thanks. Changed to use smart_open's open function instead.